src/shared/utils/types/filter.ts

```diff
@@ -0,0 +1,3 @@
+export interface IFilter<ItemProp, Result> {
+  filter: (items: ItemProp[]) => Result;
+}
```

Comment:
I suggest changing it to this:
```
export interface IFilter<TItem, TResult = TItem[]> {
  filter(items: TItem[]): TResult;
}
```

1. Generic types are conventionally prefixed by "T"
2. `TResult` should be optional
3. Unless one wants to provide their own `TResult` type, the type should be the same `ItemProp[]`
4. I made it a class method, while yours is a property storing a function.

Reply:
Done